okhttp/src/main/java/io/grpc/okhttp/OkHttpClientStream.java

```diff
@@ -372,7 +373,8 @@ private void cancel(Status reason, boolean stopDelivery, Metadata trailers) {
         requestHeaders = null;
         pendingData.clear();
         canStart = false;
-        transportReportStatus(reason, true, trailers != null ? trailers : new Metadata());
+        transportReportStatus(
+            reason, MISCARRIED, true, trailers != null ? trailers : new Metadata());
```

Comment:
So this is a case where cancel() is called before start()? I don't think we need retries at all for that case. Seems best to keep this as-it-was.